third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandWMMA.cpp

```diff
@@ -141,16 +171,15 @@ Value convertLayout(int opIdx, ConversionPatternRewriter &rewriter,
   auto repB = numReps[0];
 
   unsigned iWaveSize = triton::gpu::getWarpSize(wmmaLayout);
-  unsigned iNumLanes = iWaveSize / 2;
   assert(iWaveSize == 32);
   Value waveSize = i32_val(iWaveSize);
-  Value numLanes = i32_val(iNumLanes);
   Value linearWaveId = udiv(thread, waveSize);
-  Value lane = urem(thread, numLanes); // share elem between two threads
 
-  unsigned numElemsPerThreadPerRep = wmmaInstrK;
+  unsigned numElemsPerThreadPerRep =
+      wmmaLayout.getSizePerThreadForOperands(opIdx)[kDimIdx];
 
   Value warp = udiv(thread, waveSize);
```

Comment:
This `warp` is the same as `linearWaveId`?

Reply:
removed

Reply:
Did you forget to remove it actually? It's still here.

Reply:
Forgot to push :facepalm: